masto-module/src/main/kotlin/au/rakka/java/masto-api/MastoReplyGuy.kt

```diff
@@ -0,0 +1,92 @@
+	private val client: HttpClient = HttpClient.newHttpClient()
+	private val builder: HttpRequest.Builder = HttpRequest.newBuilder().setHeader("Authorization","Bearer ${conf.token}").setHeader("Content-Type","application/json").setHeader("User-Agent",userAgent)
+	
+	private val baseurl = "https://${conf.instance}/api/v1/" // Well it was half useful
+	private val notifurl = "${baseurl}notifications"
+	private val clearurl = "${baseurl}notifications/clear"
+	private val statusurl = "${baseurl}statuses"
+	
+	override val schedule: Int = 30
+	override suspend fun task()
+	{
+		logger.debug("Polling for masto notifs")
+		var finished=false
+		var max_id=""
+		while (!finished)
+		{
+			val response = client.sendAsync(builder.uri(URI.create(notifurl+max_id)).GET().build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+			logger.debug(response.statusCode().toString())
+			val node=mapper.readTree(response.body())
+			node.forEach{ process_post(it) }
+			logger.debug(node.size().toString())
+			finished=node.size()<20 // If node size isn't 20 I don't need to update max_id at all. Mainly avoids the issue where node size is 0.
+			if (!finished) {max_id=node.get(19).get("id").asText()}
+		}
+		val response = client.sendAsync(builder.uri(URI.create(clearurl)).POST(HttpRequest.BodyPublishers.ofString("")).build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+		logger.debug("Cleared notifications (hopefully) {}",response.statusCode().toString())
+	}
+	private suspend fun process_post(post:JsonNode)
+	{
+		logger.debug("Processing notif {}",post.get("id"))
+		if (post.get("type").asText()!="mention") {logger.debug("Was not a mention");return}
+		val text= post.get("status").get("pleroma").get("content").get("text/plain").asText()
+		val (tag,regex)=process_post_contents(text)
+		if (tag=="") {logger.debug("tag: {} was null",tag);return}
+		if (regex=="")
+		{
+			logger.debug("Deleting {}",tag)
+			wc3GameNotificationService.deleteNotification(tag) // lol hope you meant it
+			val response=client.sendAsync(builder.uri(URI.create(statusurl)).POST(HttpRequest.BodyPublishers.ofString("""{"status":"Unregistered ${tag}. In future this will hopefully be able to tell you what it contained.","in_reply_to_id":${post.get("status").get("id")}}""")).build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+			logger.debug("Deleted {} {}",response.statusCode().toString(),post.get("status").get("id"))
+		}
+		else
+		{
+			logger.debug("Adding {} with {}",tag,regex)
+			wc3GameNotificationService.createNotification(tag,regex)
+			val response=client.sendAsync(builder.uri(URI.create(statusurl)).POST(HttpRequest.BodyPublishers.ofString("""{"status":"Registered ${tag} with pattern ${regex}.","in_reply_to_id":${post.get("status").get("id")}}""")).build(),HttpResponse.BodyHandlers.ofInputStream()).await() // .get("id") returns a string with quotes: Deleted 400 "AotNh4gQIua2IPvPN2", so don't need to put new quotes on it.
```

Comment:
Do you need the `Deleted 400 "AotNh4gQIua2IPvPN2"` thing? Is that an example of the type of response?

Reply:
It's an example of output from the next line.